shell/platform/embedder/embedder.h

```diff
@@ -0,0 +1,102 @@
+// Copyright 2017 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include <stdbool.h>
+#include <stddef.h>
+#include <stdint.h>
+
+#if defined(__cplusplus)
+extern "C" {
+#endif
+
+#ifndef FLUTTER_EXPORT
+#define FLUTTER_EXPORT
+#endif  // FLUTTER_EXPORT
+
+#define FLUTTER_ENGINE_VERSION 1
+
+typedef enum {
+  kSuccess = 0,
+  kInvalidLibraryVersion,
+  kInvalidArguments,
+} FlutterResult;
+
+typedef enum {
+  kOpenGL,
+} FlutterRendererType;
+
+typedef void* FlutterEngine;
+
+typedef bool (*BoolCallback)(void* /* user data */);
+typedef uint32_t (*UIntCallback)(void* /* user data */);
+
+typedef struct {
+  BoolCallback make_current;
+  BoolCallback clear_current;
+  BoolCallback present;
+  UIntCallback fbo_callback;
+} FlutterOpenGLRendererConfig;
+
+typedef struct {
+  FlutterRendererType type;
+  union {
+    FlutterOpenGLRendererConfig open_gl;
+  };
+} FlutterRendererConfig;
+
+typedef enum {
+  kWindowMetrics,
+  kPointer,
+} FlutterEventType;
+
+typedef struct {
+  size_t width;
+  size_t height;
+} FlutterWindowMetricsEvent;
+
+typedef enum {
+  kCancel,
+  kUp,
+  kDown,
+  kMove,
+} FlutterPointerPhase;
+
+typedef struct {
```

Comment:
Using the `FLUTTER_ENGINE_VERSION` passed to initialization is ok, but not as good.  Consider a situation in which there are three binaries:

1. libflutter.so
2. libfoo.so
3. libbar.so

Now, imagine the foo and bar are compiled against different versions of libflutter.so.  If you rely upon the FLUTTER_ENGINE_VERSION passed at startup, then foo and bar need to coordinate as to their use of the flutter API (because they share the same init call).  However, if you pass sizes for each struct, then they can cooexist without needing to agree.

> Do you want me to remove the FLUTTER_ENGINE_VERSION as well?

I'd keep that.  It's valuable for the clients to be able to ifdef their code based on the version of libflutter.so they're compiling against.

> This will box us into not allowing ABI breaking updates. But maybe that is a good thing?

We will need to be able to make ABI breaks.  The idea is to make the ABI relatively flexible so that we can extend it without breaking it most of the time.  However, there will be times when we need to break it.